redis/tests/test_basic.rs

```diff
@@ -2015,4 +2015,23 @@ mod basic {
             .unwrap();
         assert!(info.contains("db=5"));
     }
+
+    #[test]
+    fn test_auth_command() {
+        let ctx = TestContext::new();
+        let mut con = ctx.connection();
+
+        redis::cmd("ACL")
+            .arg("SETUSER")
+            .arg("alice")
+            .arg("on")
+            .arg(">p1pp0")
+            .arg("+ACL")
+            .exec(&mut con)
+            .unwrap();
+
+        assert_eq!(con.auth(Some("alice"), "p1pp0"), Ok(()));
```

Comment:
please also test the None user scenario.

Reply:
Done

Reply:
Using the `None` option is cumbersome, and requires adding a meaningless type. This requires a more ergonomic solution.